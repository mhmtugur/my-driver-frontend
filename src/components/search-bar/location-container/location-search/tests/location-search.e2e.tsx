import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';

describe('location search box', () => {
  let page: E2EPage;
  let element: E2EElement;
  beforeEach(async () => {
    page = await newE2EPage({ html: '<location-search></location-search>' });
    element = await page.find('location-search');
  });

  it('should render', () => {
    expect(element.innerHTML).toMatchSnapshot();
  });

  it('should render location name when it is provided', async () => {
    element.setProperty('locationName', 'Munich Airport');
    await page.waitForChanges();
    let searchInput = await page.find('#searchInput');
    let inputValue = await searchInput.getProperty('value');
    expect(inputValue).toBe('Munich Airport');
  });

  it('should render search string when it is provided', async () => {
    element.setProperty('searchString', 'Ber');
    await page.waitForChanges();
    let searchInput = await page.find('#searchInput');
    let inputValue = await searchInput.getProperty('value');
    expect(inputValue).toBe('Ber');
  });

  it('should emit event on input change', async () => {
    const inputChangedSpy = await page.spyOnEvent('onLocationInputChange');
    await page.waitForChanges();
    let searchInput = await page.find('#searchInput');
    searchInput.press('b');
    await page.waitForChanges();
    expect(inputChangedSpy.events[0].detail).toBe('b');
  });

  it('should emit event on input focus', async () => {
    const inputChangedSpy = await page.spyOnEvent('onLocationInputFocus');
    await page.waitForChanges();
    let searchInput = await page.find('#searchInput');
    searchInput.focus();
    await page.waitForChanges();
    expect(inputChangedSpy.events[0]).toBeTruthy();
  });
});