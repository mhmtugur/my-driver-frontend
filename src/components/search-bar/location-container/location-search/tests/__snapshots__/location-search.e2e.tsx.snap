// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`location search box should render 1`] = `"<div class=\\"input-field\\"><i class=\\"material-icons prefix\\">location_on</i><input id=\\"searchInput\\" class=\\"validate truncate\\" required=\\"\\" type=\\"text\\" autocomplete=\\"off\\" placeholder=\\"City,airport,location... etc.\\"></div>"`;
